File: backend/app/schemas/subscription.py
"""订阅 Pydantic schemas。"""

from datetime import datetime

from pydantic import BaseModel, Field


class SubscriptionCreate(BaseModel):
    """创建订阅的请求体。"""

    tmdb_id: int = Field(..., description="TMDB ID")
    media_type: str = Field(..., description="媒体类型（movie / tv）")
    title: str = Field(..., description="媒体标题")
    poster_url: str | None = Field(None, description="海报 URL")
    year: int | None = Field(None, description="发行年份")


class SubscriptionResponse(BaseModel):
    """订阅记录的响应体。"""

    id: str
    tmdb_id: int
    media_type: str
    title: str
    poster_url: str | None = None
    year: int | None = None
    nf_subscribed: bool = False
    nf_status: str | None = None
    nf_missing_eps: int = 0
    nf_sub_id: str | None = None
    completed: bool = False
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SubscriptionSyncResult(BaseModel):
    """单条订阅同步结果。"""

    tmdb_id: int
    title: str
    action: str = "skipped"  # created / pushed_to_nf / updated / skipped
    nf_status: str | None = None
    nf_missing_eps: int = 0
    nf_subscribed: bool = False
    needs_mp_search: bool = False
    message: str = ""

File: frontend/src/types/subscription.ts
/**
 * 订阅相关类型 — 镜像 backend/app/schemas/subscription.py。
 */
import type { MediaType } from './common'

/** 创建订阅请求体 */
export interface SubscriptionCreate {
  tmdb_id: number
  media_type: MediaType
  title: string
  poster_url?: string | null
  year?: number | null
  source?: string | null
}

/** 订阅记录响应体 */
export interface Subscription {
  id: string
  tmdb_id: number
  media_type: MediaType
  title: string
  poster_url: string | null
  year: number | null
  nf_subscribed: boolean
  nf_status: string | null
  nf_missing_eps: number
  nf_sub_id: string | null
  completed: boolean
  aired_complete: boolean
  source: string | null
  created_at: string
  updated_at: string
  adjusted_missing_eps: number | null
  tmdb_aired_eps: number | null
}

/** 单条订阅同步结果 */
export interface SubscriptionSyncResult {
  tmdb_id: number
  title: string
  /** created / pushed_to_nf / updated / skipped */
  action: string
  nf_status: string | null
  nf_missing_eps: number
  nf_subscribed: boolean
  needs_mp_search: boolean
  message: string
}
